Source/JavaScriptCore/inspector/agents/InspectorDebuggerAgent.cpp

```diff
@@ -1490,28 +1517,85 @@ void InspectorDebuggerAgent::didCreateNativeExecutable(JSC::NativeExecutable& na
 
 void InspectorDebuggerAgent::willCallNativeExecutable(JSC::CallFrame* callFrame)
 {
+    auto& vm = m_debugger.vm();
+
     if (!breakpointsActive())
         return;
 
     if (m_symbolicBreakpoints.isEmpty())
         return;
 
-    auto symbol = functionName(callFrame);
-    if (symbol.isEmpty())
+    auto symbols = functionNames(vm, callFrame);
+    if (symbols.isEmpty())
         return;
 
-    auto index = m_symbolicBreakpoints.findIf([&] (const auto& symbolicBreakpoint) {
-        return symbolicBreakpoint.knownMatchingSymbols.contains(symbol);
-    });
-    if (index == notFound)
-        return;
+    for (auto& symbolicBreakpoint : m_symbolicBreakpoints) {
+        for (const auto& symbol : symbols) {
+            if (symbolicBreakpoint.matches(symbol)) {
+                ASSERT(symbolicBreakpoint.specialBreakpoint);
+
+                auto pauseData = JSON::Object::create();
+                pauseData->setString("name"_s, symbol);
+
+                breakProgram(DebuggerFrontendDispatcher::Reason::FunctionCall, WTFMove(pauseData), symbolicBreakpoint.specialBreakpoint.copyRef());
+                return;
+            }
+        }
+    }
+}
+
+void InspectorDebuggerAgent::willChangeDisplayName(JSC::JSFunction* jsFunction)
+{
+    auto& vm = m_debugger.vm();
+    ASSERT(&jsFunction->vm() == &vm);
+
+    if (auto* functionExecutable = JSC::jsDynamicCast<JSC::FunctionExecutable*>(jsFunction->executable())) {
+        if (auto symbols = functionNames(vm, *functionExecutable); !symbols.isEmpty()) {
+            for (auto& symbolicBreakpoint : m_symbolicBreakpoints) {
+                if (!symbolicBreakpoint.matches(symbols))
+                    continue;
+
+                if (auto* codeBlockForCall = functionExecutable->codeBlockFor(JSC::CodeForCall)) {
```

Comment:
@WebKit/jsc-reviewers is this the correct way to get all `CodeBlock` for a given `FunctionExecutable`?  or do I also need to do something like `baselineCodeBlockFor` (and check that the result is not the same as `codeBlockFor`)?